enhancements/multi-arch/dynamic-imagestream-importmode-setting.md

```diff
@@ -0,0 +1,408 @@
+`json:"imagestreamImportMode,omitempty"
+}
+```
+- openshiftcontrolplane/v1/: Add
+  `ImagestreamImportMode` config flag to
+  ImagePolicyConfig struct.
+
+```
+type ImagePolicyConfig {
+...
+...
+// imagestreamImportMode controls the import
+mode behaviour of imagestreams. It can be set
+to `Legacy` or `PreserveOriginal`. The
+// default behaviour is to default to Legacy.
+// If this value is specified, this setting is
+// applied to all imagestreams which do not
+// have the value set.
+// +optional
+ImagestreamImportMode imagev1.ImportModeType
+`json:"imagestreamImportMode,omitempty"`
+
+}
+```
+- An example CR for `image.config.openshift.io`
+  would look like below if the import mode is set:
+```
+apiVersion: config.openshift.io/v1
+kind: Image
+metadata:
+  name: cluster
+  ...
+  ...
+spec:
+  imagestreamImportMode: PreserveOriginal
+status:
+   internalRegistryHostname: image-registry.openshift-image-registry.svc:5000
+   imagestreamImportMode: PreserveOriginal
+  ...
+  ...
+
+```
+- The ClusterVersion status field CR would look
+  like below after CVO has inferred the payload
+  type:
+```
+apiVersion: v1
+items:
+- apiVersion: config.openshift.io/v1
+  kind: ClusterVersion
+  spec:
+    channel: candidate-4.15
+    desiredUpdate:
+      force: false
+      version: 4.15.0
+  status:
+    availableUpdates:
+    ...
+    capabilities:
+    ...
+    conditions:
+    ...
+    desired:
+    ...
+    history:
+    ...
+    payloadArchitecture: amd64  # could be arm64/ppc64le/s390x/multi
+    ...
+``` 
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+N/A
+
+#### Standalone Clusters
+
+N/A
+
+#### Single-node Deployments or MicroShift
+
+N/A
+
+### Implementation Details/Notes/Constraints
+
+- CVO: The cluster version operator would expose a
+  field in its status indicating the
+  architecture/type of the payload.
+    - If architecture, then it could be `amd64`,
+      `arm64`, `ppc64le`, `s390x` or `multi`.
+    - If type, then it could be `single` or
+      `multi`.
+- Cluster-image-registry-operator: the operator
+  sets the `imagestreamImportMode` field in the
+  status of the `images.config.openshift.io`
+  cluster image config based on whether:
+    - `images.config.openshift.io` image config
+      has importmode set in its spec.
+    - if no global importmode is set in the image
+      config, the operator sets it based on the
+      payload type (inferred from CVO).Multi would
+      mean `PreserveOriginal` and any other value
+      would mean `Legacy`(single-arch).
+- Cluster-openshift-apiserver-operator: the
+  operator sets the import mode value in the
+  OpenshiftAPIServer CR's observed config based on
+  the `imagestreamImportMode` in the image config
+  status.
+- Openshift-apiserver: The apiserver  would look
+  at the observed config, check the import mode
+  value and set the default for the imagestreams
+  based on that value.
+
+### Risks and Mitigations
+
+N/A
+
+### Drawbacks
```

Comment:
One concern I have with enabling PreserveOriginal import mode for all future image stream imports cluster-wide is the potential growth of storage space consumption in etcd.
For each sub manifest present in a manifest list, there will be an equivalent Image object with a list of its layers + other basic info. 
While Image objects aren't extremely big, a single manifest list usually has a few sub manifests in it, so instead of having just 1 Image object (like what the Legacy mode does), we will potentially have the manifest list + the number of architectures it supports. So if a manifest list supports 4 architectures, there would be a total of 5 Image objects created in etcd... for just one manifest listed tag imported.

Reply:
Does this depend on whether the reference policy is local or from source? in other words even if it is from source, do we incur the cost of these entries? I'll make a note of this.

Reply:
Yes, the reference policy is about whether the image registry itself stores copies of the images or not. Image objects are always stored [in etcd] regardless of which reference policy is used.

Reply:
added this to the risks section